### ISSUE TITLE
Webcam motion detection

### ISSUE THREAD

**Issue by Lvl4Sword** [Lvl4Sword/Killer#48]
Continuation of guardianproject/haven#390

**Comment by Lvl4Sword** [comment]
@HulaHoopWhonix
``For motion detection there is "motion" on Linux that takes snapshots when it detects movement. Maybe killer can watch its directory for any newly created files and kill when any are created.
https://www.maketecheasier.com/setup-motion-detection-webcam-ubuntu/

For light detection you may be able to repurpose the guts of autobrightness/Wildguppy to measure light changes, but instead of adjusting display brightness you would shut off the system.``

Shutting the system off right now is default, that's not going to be forced in the future and will be completely optional.

**Comment by HulaHoopWhonix** [comment]
> Shutting the system off right now is default, that's not going to be forced in the future and will be completely optional.

Even better! Interesting so custom commands as in sending IM alerts for example? (or anything a sysadmin can plumb together with a scripts?)

***
Thanks for considering this idea. I guess I hadn't thought out that killer can natively cover most of the same ground as Haven does, but on PCs.

Please keep us updated and we would like to at least include it in our repos if not in Debian proper.

**Comment by Lvl4Sword** [comment]
>Even better! Interesting so custom commands as in sending IM alerts for example? (or anything a sysadmin can plumb together with a scripts?)

Yes, but custom commands won't be possible until 2.0 is released, and https://github.com/Lvl4Sword/Killer/issues/47#issuecomment-480390187 covers as to what defines 1.0/2.0 ( though it does need fleshed out a bit more ) so we have a long way to go.

**Comment by Lvl4Sword** [comment]
For discussing this or anything else, there's also an IRC and Discord linked within the README if you'd like to join and talk there.

**Comment by GhostofGoes** [comment]
Another related source would be a "presence sensor" that are on some newer devices (mainly laptops), as well as possibly using light sensors.

**Comment by Lvl4Sword** [comment]
@GhostofGoes Never heard of a laptop presence sensor.
Should probably be put in it's own issue.
